iceoryx_posh/include/iceoryx_posh/internal/popo/ports/client_port_roudi.hpp

```diff
@@ -36,14 +36,20 @@ class ClientPortRouDi : public BasePort
   public:
     using MemberType_t = ClientPortData;
 
-    explicit ClientPortRouDi(cxx::not_null<MemberType_t* const> clientPortDataPtr) noexcept;
+    /// @brief Creates a ClientPortRouDi from ClientPortData which are shared with ClientPortUser
+    /// @param[in] clientPortData to be are accessed by the ClientPortRouDi interface
+    explicit ClientPortRouDi(MemberType_t& clientPortData) noexcept;
 
     ClientPortRouDi(const ClientPortRouDi& other) = delete;
     ClientPortRouDi& operator=(const ClientPortRouDi&) = delete;
     ClientPortRouDi(ClientPortRouDi&& rhs) = default;
     ClientPortRouDi& operator=(ClientPortRouDi&& rhs) = default;
     ~ClientPortRouDi() = default;
 
+    /// @brief Access to the configured responseQueueFullPolicy
+    /// @return the configured responseQueueFullPolicy
+    QueueFullPolicy2 getResponseQueueFullPolicy() const noexcept;
```

Comment:
The current `QueueFullPolicy` will later on be replaced by `QueueFullPolicy2` (and renamed to `QueueFullPolicy`). Since it is a breaking change and touches quite some files and examples. This will be done at the end of the feature.

From `port_queue_policies.hpp`
```cpp
/// @brief Used by subscriber
/// @todo iox-#27 replace with QueueFullPolicy2
enum class QueueFullPolicy : uint8_t
{
    BLOCK_PUBLISHER,
    DISCARD_OLDEST_DATA
};

...

/// @brief Used by consumers to request a specific behavior from the producer
/// @todo iox-#27 Rename this to QueueFullPolicy
enum class QueueFullPolicy2
{
    /// Requests the producer to block when the consumer queue is full
    BLOCK_PRODUCER,
    /// Request to discard the oldest data and push the newest one into the queue
    DISCARD_OLDEST_DATA
};
```